lib/db/schema/patch-086-087.sql

```diff
@@ -0,0 +1,262 @@
+SET NAMES utf8mb4 COLLATE utf8mb4_bin;
+
+CALL assertPatchLevel('86');
+
+-- Add the `profileChangedAt` column to the accounts table.
+-- Default NULL implies that the account profile hasn't changed since it
+-- was created.
+ALTER TABLE accounts ADD COLUMN profileChangedAt BIGINT UNSIGNED DEFAULT NULL,
+ALGORITHM = INPLACE, LOCK = NONE;
+
+-- Update `profileChangedAt` when emails are changed, verified or deleted
+CREATE PROCEDURE `setPrimaryEmail_2` (
+  IN `inUid` BINARY(16),
+  IN `inNormalizedEmail` VARCHAR(255)
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+     UPDATE emails SET isPrimary = false WHERE uid = inUid AND isPrimary = true;
+     UPDATE emails SET isPrimary = true WHERE uid = inUid AND isPrimary = false AND normalizedEmail = inNormalizedEmail;
+
+     SELECT ROW_COUNT() INTO @updateCount;
+     IF @updateCount = 0 THEN
+        SIGNAL SQLSTATE '45000' SET MYSQL_ERRNO = 1062, MESSAGE_TEXT = 'Can not change email. Could not find email.';
+     END IF;
+
+     UPDATE accounts SET profileChangedAt = (UNIX_TIMESTAMP(NOW(3)) * 1000) WHERE uid = inUid;
+  COMMIT;
+END;
+
+CREATE PROCEDURE `verifyEmail_6`(
+    IN `inUid` BINARY(16),
+    IN `inEmailCode` BINARY(16)
+)
+BEGIN
+    DECLARE EXIT HANDLER FOR SQLEXCEPTION
+    BEGIN
+        ROLLBACK;
+        RESIGNAL;
+    END;
+
+    START TRANSACTION;
+
+    UPDATE accounts SET emailVerified = true WHERE uid = inUid AND emailCode = inEmailCode;
+    UPDATE emails SET isVerified = true WHERE uid = inUid AND emailCode = inEmailCode;
+
+    UPDATE accounts SET profileChangedAt = (UNIX_TIMESTAMP(NOW(3)) * 1000) WHERE uid = inUid;
+
+    COMMIT;
+END;
+
+CREATE PROCEDURE `deleteEmail_3` (
+    IN `inUid` BINARY(16),
+    IN `inNormalizedEmail` VARCHAR(255)
+)
+BEGIN
+    SET @primaryEmailCount = 0;
+
+    -- Don't delete primary email addresses
+    SELECT COUNT(*) INTO @primaryEmailCount FROM emails WHERE normalizedEmail = inNormalizedEmail AND uid = inUid AND isPrimary = true;
+    IF @primaryEmailCount = 1 THEN
+        SIGNAL SQLSTATE '45000' SET MYSQL_ERRNO = 2100, MESSAGE_TEXT = 'Can not delete a primary email address.';
+    END IF;
+
+    DELETE FROM emails WHERE normalizedEmail = inNormalizedEmail AND uid = inUid AND isPrimary = false;
+
+    UPDATE accounts SET profileChangedAt = (UNIX_TIMESTAMP(NOW(3)) * 1000) WHERE uid = inUid;
+END;
+
+-- Update `profileChangedAt` when TOTP is only verified or deleted on an account
```

Comment:
I chose to not update the property when a TOTP token is created because it doesn't get "turned on" until after a call to `updateTotpToken`. An unverified token would not have changed anything on the account.

Reply:
It may be worth saying what you say here in a comment in github, in the comment in the code for posterity.